BnbnavNetClient/Views/MapView.axaml.cs

```diff
@@ -90,37 +91,70 @@ protected override void OnInitialized()
                 _toScreenMtx = matrix;
                 _toWorldMtx = matrix.Invert();
             }));
+
+
         MapViewModel
             .WhenAnyPropertyChanged()
             .Subscribe(Observer.Create<MapViewModel?>(_ => { InvalidateVisual(); }));
     }
 
-    static readonly double NodeSize = 14;
     static readonly double LandmarkSize = 10;
-    static readonly IPen BlackBorderPen = new Pen(new SolidColorBrush(Colors.Black), thickness: 2);
-    static readonly IBrush BackgroundBrush = new SolidColorBrush(Colors.WhiteSmoke);
-    static readonly IBrush WhiteFillBrush = new SolidColorBrush(Colors.White);
-    static readonly Pen RoadPen = new Pen(new SolidColorBrush(Colors.DarkBlue), thickness: 20, lineCap: PenLineCap.Round);
-
     readonly Dictionary<string, SKSvg> _svgCache = new();
 
     public override void Render(DrawingContext context)
     {
         var mapService = MapViewModel.MapService;
         var scale = MapViewModel.Scale;
 
-        context.FillRectangle(BackgroundBrush, Bounds);
+        context.FillRectangle((Brush)this.FindResource("BackgroundBrush")!, Bounds);
 
         foreach (var edge in mapService.Edges.Values)
         {
+            var pen = (Pen)(edge.Road.RoadType switch
+            {
+                RoadType.Local => this.FindResource("LocalRoadPen")!,
+                RoadType.Main => this.FindResource("MainRoadPen")!,
+                RoadType.Highway => this.FindResource("HighwayRoadPen")!,
+                RoadType.Expressway => this.FindResource("ExpresswayRoadPen")!,
+                RoadType.Motorway => this.FindResource("MotorwayRoadPen")!,
+                RoadType.Footpath => this.FindResource("FootpathRoadPen")!,
+                RoadType.Waterway => this.FindResource("WaterwayRoadPen")!,
+                RoadType.Private => this.FindResource("PrivateRoadPen")!,
+                RoadType.Roundabout => this.FindResource("RoundaboutRoadPen")!,
+                RoadType.DuongWarp => this.FindResource("DuongWarpRoadPen")!,
+                _ => this.FindResource("UnknownRoadPen")!,
+            });  
             var fromEdge = mapService.Nodes[edge.From.Id];
             var from = ToScreen(new(fromEdge.X, fromEdge.Z));
             var toEdge = mapService.Nodes[edge.To.Id];
             var to = ToScreen(new (toEdge.X, toEdge.Z));
+
+            static double Angle(Point a, Point b)
```

Comment:
Are these functions required? We only use them once so maybe we can inline them, or maybe they're useful elsewhere so maybe we can take it out into a helper class or something?